config/file_test.go

```diff
@@ -340,37 +380,20 @@ func TestFileStoreGetEnivironmentOverrides(t *testing.T) {
 }
 
 func TestFileStoreSet(t *testing.T) {
-	t.Run("set same pointer value", func(t *testing.T) {
-		t.Skip("not yet implemented")
-
-		path, tearDown := setupConfigFile(t, emptyConfig)
-		defer tearDown()
-
-		fs, err := config.NewFileStore(path, false)
-		require.NoError(t, err)
-		defer fs.Close()
-
-		_, err = fs.Set(fs.Get())
-		if assert.Error(t, err) {
-			assert.EqualError(t, err, "old configuration modified instead of cloning")
-		}
-	})
-
 	t.Run("defaults required", func(t *testing.T) {
 		path, tearDown := setupConfigFile(t, minimalConfig)
 		defer tearDown()
 
-		fs, err := config.NewFileStore(path, false)
+		fsiner, err := config.NewFileStore(path, false)
+		require.NoError(t, err)
+		fs, err := config.NewStoreFromBacking(fsiner)
 		require.NoError(t, err)
 		defer fs.Close()
 
-		oldCfg := fs.Get()
-
 		newCfg := &model.Config{}
 
-		retCfg, err := fs.Set(newCfg)
+		_, err = fs.Set(newCfg)
```

Comment:
Can you clarify why we are not checking the old config anymore? It's not very clear to me.

Reply:
Because we used to not add defaults here, but now we do because the file stores are always part of the config.Store.

Reply:
Ah gotcha, so in the actual code, a wrapped store will add the defaults and that's what we expect. But a pure file store is not supposed to do that in a test?